linux/device.go

```diff
@@ -449,8 +449,8 @@ func createLinuxDevice(volume *model.Volume) (dev *model.Device, err error) {
 	// cleanup paths which maybe part of lsscsi but not in multipath as we haven't found the device yet
 	cleanupStaleScsiPaths(volume)
 	// try to logout the iscsi target if we could not find a device by now for VST Volume
-	if volume.TargetScope != GroupScope.String() && volume.Iqn != "" {
-		iscsiLogoutOfTarget(&model.IscsiTarget{Name: volume.Iqn})
+	if volume.TargetScope != GroupScope.String() && len(volume.TargetNames()) != 0 {
+		iscsiLogoutOfTarget(&model.IscsiTarget{Name: volume.TargetNames()[0]})
```

Comment:
What if there are more than 1?  Don't we have to log out of all targets?

Reply:
no, we logout only for VST volumes. with GST as other devices are sharing it we never logout.

Reply:
But won't this code be shared with 3Par?  Does that apply to them too?

Reply:
yes, 3par is GST by default. They should be setting TargetScope as group. They can be logged-in always without any devices and doesn't cause any issues in linux.

Reply:
Cool.  Thanks.

Reply:
BTW, they don't set target_scope.  It's assumed to be group if not provided.

Reply:
got it..yes that is good.